### PR TITLE
Adds Makefile Targets for API and UI and Updates the pre-submit PR script

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
- This patch adds a Makefile targets for

     - API: Unit Test, Lint, Build, Update Golden files
     - UI: Unit Test, Lint, Build
     - Yamllint check
     
  ```
$ make help    
test                 run all Unit Tests
lint                 check all Lints
api-check            all API checks
ui-check             all UI checks
goa-gen              generate API Design
generated            update the golden files
api-lint             run API Lint
api-test             run API Unit Test
api-build            generate the API binary
ui-lint              run UI Lint
ui-test              run UI Unit Test
ui-build             generate the UI binary
yaml-lint            run YAML Lint
help                 print this help
```

Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/main/CONTRIBUTING.md) for more details._

### CODE REVIEW

**Comment by sm43** [comment]
could you document the command available through make file and what they do?

**Comment by PuneetPunamiya** [comment]
> could you document the command available through make file and what they do?

Where that file should be located ??

**Comment by sm43** [comment]
> > could you document the command available through make file and what they do?
> 
> Where that file should be located ??

Ideally Development.md
also if we have a command which does goa gen - run unit test - lint  and everything needed before creating pr 
we can add that in pr template similar to tkn cli

**Comment by PuneetPunamiya** [comment]
> > > could you document the command available through make file and what they do?
> > 
> > 
> > Where that file should be located ??
> 
> Development.md
> also if we have a command which does goa gen - run unit test - lint and everything needed before creating pr
> we can add that in pr template similar to tkn cli

Sounds good will update the pr thanks 👍🏻

File: test/presubmit-tests.sh [hunk 1]
```diff
@@ -54,10 +54,9 @@ install-node() {
 }
 
 ui-unittest(){
-    install-node
-    npm clean-install
+  install-node
 
-  CI=true npm test || {
+  make ui-test || {
```
Comment (vinamra28) [inline test/presubmit-tests.sh:59]: why are we removing `CI=true` ?

File: test/presubmit-tests.sh [hunk 2]
```diff
@@ -87,50 +86,39 @@ api-unittest(){
...
 }
 
 ui-build(){
-  info Running UI build
-
   install-node
-  npm clean-install
 
-  CI=true npm run build || {
+  make ui-build || {
```
Comment (vinamra28) [inline test/presubmit-tests.sh:121]: same as above

File: test/presubmit-tests.sh (hunk 1 above)
Comment (PuneetPunamiya) [reply test/presubmit-tests.sh:59]: I have added that in the specified make target
For e.g.
```
# Make target for UI Unit Test
.PHONY: ui-test
ui-test:
        @echo "----------------------------"
        @echo "-- Running UI Unit Tests --"
        @echo "----------------------------"
        cd ui && npm clean-install && CI=true npm test
```

**Comment by vinamra28** [comment]
should we add `make help` as well which can list all possible subcommands?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -11,7 +11,8 @@ These are the criteria that every PR should meet, please check them off as you
 review them:
 
 - [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
+- [ ] Run API Unit Tests, Lint Checks, API Design, Golden Files with `make api-check`
+- [ ] Run API Unit Tests, Lint Checks with `make ui-check`
```
Comment (sm43) [inline .github/pull_request_template.md:15]: nit: `UI`

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+goa-gen: ## generate API Design
+	@echo "----------------------------"
+	@echo "-- Generating API Design... "
+	@echo "----------------------------"
+	cd api && goa gen github.com/tektoncd/hub/api/design
+
+	@echo "----------------------------"
+	@echo "- Generating v1 API Design... "
+	@echo "----------------------------"
+	cd api && cd v1 && goa gen github.com/tektoncd/hub/api/v1/design
```
Comment (sm43) [inline Makefile:23]: how does this work?
`cd api && goa gen github.com/tektoncd/hub/api/design`
when we do this.. does the directory context changes only for that command?
and 
again below it does
`cd api/v1` so each command runs in  a separate context/terminal?

Comment (PuneetPunamiya) [reply Makefile:23]: Yes the directory content changes only for that command and also the command runs in a separate context

File: api/update.sh [hunk 5]
```diff
@@ -0,0 +1 @@
+go test $(go list -f '{{ .ImportPath }} {{ .TestImports }}' ./... | grep gotest.tools/v3/golden | awk '{print $1}' | tr '\n' ' ') -test.update-golden=true
```
Comment (sm43) [inline api/update.sh:1]: does this requires to be in separate file?

Comment (vinamra28) [reply api/update.sh:1]: yeah I guess.... we can use it independently as well

**Review by sm43 — CHANGES_REQUESTED** [review]

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+	@echo "- Generating v1 API Design... "
+	@echo "----------------------------"
+	cd api/v1 && goa gen github.com/tektoncd/hub/api/v1/design
+
+.PHONY: generated
+generated: ## update the golden files
+	@echo "----------------------------"
+	@echo "-- Generating Golden Files... --"
+	@echo "----------------------------"
+	cd api && go mod vendor && bash update.sh
```
Comment (vinamra28) [inline Makefile:30]:
```suggestion
	cd api && go mod vendor && bash update-golden-files.sh
```

**Pull request merged**